Conceptpower+Spring/src/main/java/edu/asu/conceptpower/app/db/DBRequestClient.java

```diff
@@ -48,5 +63,49 @@ public ReviewRequest getReviewRequestForConcept(String conceptId){
         }
         return null;
     }
-   
+    
+    @Override
+    public ReviewRequest updateReviewRequest(ReviewRequest reviewRequest) {
+        
+        ReviewRequest requestToBeUpdated = client.query(new Predicate<ReviewRequest>() {
+            private static final long serialVersionUID = 6495914730735826451L;
+
+            @Override
+            public boolean match(ReviewRequest review) {
+                return review.getConceptId().equals(reviewRequest.getConceptId());
```

Comment:
you can just query by example, no predicate needed (it's usually really slow). you should also be able to index by concept id, which should speed it up.

Reply:
Using queryByExample will simply return a object but it won't point to the exact row value.

So updating that object and saving it back will create an another entry. Using predicate should avoid that. Correct me if I'm wrong.

Reply:
actually, I might have been wrong. predicates might be using indexes and might be optimized. let's leave it like that, since we'll be moving to MySql anyways.